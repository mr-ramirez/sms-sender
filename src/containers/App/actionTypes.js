const ActionTypes = {
  FINISH_LOADING_ALL_MEMBERS: 'FINISH_LOADING_ALL_MEMBERS',
  SET_MEMBERS: 'SET_MEMBERS',
  START_LOADING_ALL_MEMBERS: 'START_LOADING_ALL_MEMBERS',
};

export default ActionTypes;
