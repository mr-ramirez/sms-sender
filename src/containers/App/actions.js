import ActionTypes from './actionTypes';
import HttpService from '../../api/HttpService';
import Member from '../../models/member';

export function startLoadingAllMembers() {
  return {
    type: ActionTypes.START_LOADING_ALL_MEMBERS,
  };
}

export function finishLoadingAllMembers() {
  return {
    type: ActionTypes.FINISH_LOADING_ALL_MEMBERS,
  };
}

export function setMembers(members) {
  return {
    type: ActionTypes.SET_MEMBERS,
    members
  };
}

export function loadAllMembers({ chamber, congress }) {
  return (dispatch) => {
    dispatch(startLoadingAllMembers());

    return HttpService.get(`${congress}/${chamber}/members.json`)
      .then((response) => {
        const { results } = response;
        const resultsMapped = results[0].members.map(item => new Member(item));

        dispatch(setMembers(resultsMapped));
        return dispatch(finishLoadingAllMembers());
      })
      .catch(() => dispatch(finishLoadingAllMembers()));
  };
}
