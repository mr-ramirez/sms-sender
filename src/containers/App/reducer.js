import ActionTypes from './actionTypes';

const initialState = {
  loading: false,
  members: [],
};

function appReducer(state = initialState, action) {
  const { type } = action;

  switch(type) {
  case ActionTypes.FINISH_LOADING_ALL_MEMBERS:
    return {
      ...state,
      loading: false,
    };
  case ActionTypes.SET_MEMBERS:
    return {
      ...state,
      loading: true,
      members: action.members,
    };
  case ActionTypes.START_LOADING_ALL_MEMBERS:
    return {
      ...state,
      loading: true,
    };
  default:
    return state;
  }
}

export default appReducer;
