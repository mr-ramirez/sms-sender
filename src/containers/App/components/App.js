import { connect } from 'react-redux';
import React, { Component } from 'react';
import PropTypes from 'prop-types';

import '../App.css';
import Header from '../../../components/Header';

class App extends Component {
  constructor() {
    super();

    this.state = {
      members: [],
    };
  }

  render() {
    return (
      <div className="App">
        <Header />

        <section className="container">
          <h1>Boilerplate ready</h1>
        </section>
      </div>
    );
  }
}

App.propTypes = {
  app: PropTypes.object.isRequired,
};

function mapStateToProps(state) {
  return {
    ...state,
  };
}

function mapDispatchToProps(dispatch) {
  return {
    load: () =>
      dispatch(console.info('Loading')), // eslint-disable-line no-console
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
