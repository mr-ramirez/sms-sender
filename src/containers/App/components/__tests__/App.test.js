import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';

jest.mock('../../../../components/Spinner', () => 'Spinner');

describe('App.js', () => {
  const mockedStore = {
    dispatch: jest.fn(),
    getState: jest.fn(),
    subscribe: jest.fn(),
  };

  const mockedPropsApp = {
    loading: true,
    members: [{
      id: 10,
    }],
  };

  it('THEN it should render with a spinner and the search component', () => {
    const component = renderer.create(
      <App
        app={mockedPropsApp}
        store={mockedStore} />,
    );
    const tree = component.toJSON();
    
    expect(tree).toMatchSnapshot();
  });
});
