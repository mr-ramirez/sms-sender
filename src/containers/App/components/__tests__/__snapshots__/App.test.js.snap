// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.js THEN it should render component 1`] = `
<div
  className="App"
>
  <div>
    <header
      className="Header_container"
    >
      <img
        alt="logo"
        className="Header_logo"
        src="logo.png"
      />
      <h1
        className="Header_title"
      >
        React Programming Exercise
      </h1>
    </header>
    <div>
      <nav
        className="navbar navbar-expand-lg navbar-light bg-light"
      >
        <button
          aria-controls="navbarNav"
          aria-expanded="false"
          aria-label="Toggle navigation"
          className="navbar-toggler"
          data-target="#navbarNav"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          id="navbarNav"
        >
          <ul
            className="navbar-nav"
          >
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/"
                onClick={[Function]}
              >
                Home
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </div>
  </div>
  <section
    className="container"
  >
    <h1>
      Boilerplate ready
    </h1>
  </section>
</div>
`;
